docs/development.rst

```diff
@@ -26,157 +121,606 @@ Quickstart
+To bypass the checks (to save or get feedback on work-in-progress) use
+``git commit --no-verify``
+
+.. _sec_development_documentation:
+
+*************
+Documentation
+*************
 
-**********
-Code style
-**********
+The documentation for tskit is written using
+`Sphinx <http://www.sphinx-doc.org/en/stable/>`_
+and contained in the ``docs`` directory. It is written in the
+`reStructuredText <http://docutils.sourceforge.net/rst.html>`_ format and
+deployed automatically to `readthedocs <https://readthedocs.org/>`_.
+API documentation for both Python and C are generated automatically from
+source.
+For the C code, a combination of `Doxygen <http://www.doxygen.nl/>`_
+and `breathe <https://breathe.readthedocs.io/en/latest/>`_ is used to
+generate API documentation.
 
-Submitted python code should conform to `PEP8 <https://www.python
-.org/dev/peps/pep-0008/>`_. `Black <https://github.com/psf/black>`_ and
-`flake8 <https://flake8.pycqa.org/en/latest/>`_ are used as part of the pre-commit
-hook for python code style and formatting.
+To build the documentation locally, ``cd`` into the ``docs``
+directory and run ``make``. This should build the HTML
+documentation in ``docs/_build/html/``, which you can then view in
+your browswer.
 
-C code is formatted using `clang-format` with a custom
-configuration: `clang-format -i c/tskit/* c/tests/*.c c/tests/*.h`
+
+.. _sec_development_python:
+
+**************
+Python library
+**************
+
+The Python library is defined in the ``python`` directory. We assume throughout
+this section that you have ``cd``'d into this directory.
+We also assume that the ``tskit`` package is built and
+run locally *within* this directory. That is, ``tskit`` is *not* installed
+into the Python installation using ``pip install -e`` or setuptools `development
+mode <http://setuptools.readthedocs.io/en/latest/setuptools.html#id23>`_.
+Please see the :ref:`sec_development_python_troubleshooting` section for help
+if you encounter problems with compiling or running the tests.
+
+---------------
+Getting started
+---------------
+
+After you have installed the basic :ref:`sec_development_getting_started_requirements`
+and created a :ref:`development environment <sec_development_getting_started_environment>`,
+you will need to compile the low-level :ref:`sec_development_python_c` module.
+This is most easily done using ``make``:
+
+.. code-block:: bash
+
+    $ make
+
+If this has completed successfully you should see a file ``_tskit.cpython-XXXXXX.so``
+in the current directory (the suffix depends on your platform and Python version;
+with Python 3.6 on linux it's ``_tskit.cpython-36m-x86_64-linux-gnu.so``).
+
+To make sure that your development environment is working, run some tests:
+
+.. code-block:: bash
+
+    $ python3 -m nose -vs tests/test_tables.py
+
+These should all pass. (We've only run a subset of the tests because running the
+full suite takes few minutes.)
```

Comment:
should say how to run them all - e.g. "We've only run a subset of the tests because running the
full suite takes few minutes; to run them all run `python3 -m nose -vs tests`."

Reply:
Hm, maybe this should be merged with the "Tests" section below?